genai_factory/src/genai_factory/workflows/workflow.py

```diff
@@ -0,0 +1,144 @@
+# Copyright 2023 Iguazio
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from typing import List, Union
+
+import mlrun.serving as mlrun_serving
+from mlrun.utils import get_caller_globals
+
+from genai_factory.config import WorkflowServerConfig
+from genai_factory.schemas import APIDictResponse, WorkflowType
+from genai_factory.schemas import Workflow as WorkflowSchema
+from genai_factory.sessions import SessionStore
+
+
+class Workflow:
+    def __init__(
+        self,
+        name: str,
+        version: str,
+        workflow_type: WorkflowType,
+        skeleton: Union[List[Union[mlrun_serving.states.FlowStep, dict]], dict],
+        session_store: SessionStore,
+        config: WorkflowServerConfig,
+        description: str = "",
+        labels: dict = None,
+        deployment: str = None,
+    ):
+        # Validate the skeleton:
+        if not skeleton:
+            raise ValueError("The workflow skeleton must not be empty")
+
+        # Store parameters:
+        self._name = name
+        self._version = version
+        self._workflow_type = workflow_type
+        self._skeleton = skeleton
+        self._session_store = session_store
+        self._config = config
+        self._labels = labels
+        self._description = description
+        self._deployment = deployment
+
+        # Prepare future instances:
+        self._graph = None
+        self._server = None
+
+    def to_schema(self) -> WorkflowSchema:
+        return WorkflowSchema(
+            name=self._name,
+            version=self._version,
+            workflow_type=self._workflow_type,
+            configuration=self.get_config(),
+            graph=self._graph.to_dict(),
+            labels=self._labels,
+            description=self._description,
+            deployment=self._deployment,
+        )
+
+    def get_config(self):
+        return self._config.workflows_kwargs.get(self._name, {})
+
+    def build(self):
+        config = self.get_config()
```

Comment:
It is a bit misleading this `get_config`... Let's do it this way instead:

```suggestion
    def build(self):
        steps_config = self._config.workflows_kwargs.get(self._name, {}).get("steps", {})
```